src/components/Footer.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  setVisibleTodos: (todos: Todo[]) => void;
+  clearCompletedTodos: () => void;
+};
+
+enum Status {
+  All = 'All',
+  Active = 'Active',
+  Completed = 'Completed',
+}
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  setVisibleTodos = () => { },
+  clearCompletedTodos = () => { },
+}) => {
+  const [filter, setFilter] = useState<Status>(Status.All);
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const handlerFilterChange = (status: Status) => {
+    setFilter(status);
+    switch (status) {
+      case Status.Active:
+        setVisibleTodos(activeTodos);
+        break;
+      case Status.Completed:
+        setVisibleTodos(completedTodos);
+        break;
+      default:
+        setVisibleTodos(todos);
+    }
+  };
+
+  const leftTodos = todos.filter(todo => !todo.completed).length;
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {leftTodos === 1 ? (
+          '1 items left'
+        ) : (
+          `${leftTodos} items left`
+        )}
+      </span>
+
+      {/* Active filter should have a 'selected' class */}
```

Comment:
Remove comment. Try to write obvious code with meaningful variable and function naming. Fix all cases